core/include/scipp/core/element/util.h

```diff
@@ -41,4 +42,24 @@ template <class T> void zero(const core::ValueAndVariance<span<T>> &data) {
   zero(data.variance);
 }
 
+constexpr auto values =
+    overloaded{transform_flags::no_out_variance,
+               core::element::arg_list<double, float>, [](const auto &x) {
+                 if constexpr (is_ValueAndVariance_v<std::decay_t<decltype(x)>>)
+                   return x.value;
+                 else
+                   return x;
```

Comment:
Minor comment - This is also returning `unit` might be nicer to have a separate lambda for that as below.